homeassistant/components/ecobee/switch.py

```diff
@@ -0,0 +1,98 @@
+"""Support for using switch with ecobee thermostats."""
+
+from __future__ import annotations
+
+from datetime import datetime
+import logging
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import EcobeeData
+from .const import DOMAIN
+from .entity import EcobeeBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ecobee thermostat switch entity."""
+    data: EcobeeData = hass.data[DOMAIN]
+
+    async_add_entities(
+        (
+            EcobeeVentilator20MinSwitch(data, index)
+            for index, thermostat in enumerate(data.ecobee.thermostats)
+            if thermostat["settings"]["ventilatorType"] != "none"
+        ),
+        True,
+    )
+
+
+class EcobeeVentilator20MinSwitch(EcobeeBaseEntity, SwitchEntity):
+    """A Switch class, representing 20 min timer for an ecobee thermostat with ventilator attached."""
+
+    _attr_has_entity_name = True
+    _attr_name = "Ventilator 20m Timer"
+
+    def __init__(
+        self,
+        data: EcobeeData,
+        thermostat_index: int,
+    ) -> None:
+        """Initialize ecobee ventilator platform."""
+        super().__init__(data, thermostat_index)
+        self._attr_unique_id = f"{self.base_unique_id}_ventilator_20m_timer"
+        self._attr_is_on = False
+        self.update_without_throttle = False
+
+    @property
+    def is_on(self) -> bool | None:
```

Comment:
We can remove this property completely. The base switch class will use the `_attr_is_on` attribute to set state.